resources/turso/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+use async_trait::async_trait;
+use libsql_client::{client::Client, new_client_from_config, Config};
+use serde::{Deserialize, Serialize};
+use shuttle_service::{
+    error::{CustomError, Error as ShuttleError},
+    Factory, ResourceBuilder, Type,
+};
+use url::Url;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct Turso {
+    addr: String,
+    token: String,
+    local_addr: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct TursoOutput {
+    conn_url: Url,
+    token: Option<String>,
+}
+
+impl Turso {
+    pub fn local_addr(mut self, local_addr: &str) -> Self {
+        self.local_addr = Some(local_addr.to_string());
+        self
+    }
+}
+
+pub enum Error {
+    UrlParseError(url::ParseError),
+}
+
+impl From<Error> for shuttle_service::Error {
+    fn from(error: Error) -> Self {
+        let msg = match error {
+            Error::UrlParseError(err) => format!("Cannot parse Turso Url: {}", err),
+        };
+
+        ShuttleError::Custom(CustomError::msg(msg))
+    }
+}
+
+#[async_trait]
+impl ResourceBuilder<Client> for Turso {
+    const TYPE: Type = Type::Turso;
+
+    type Config = Self;
+    type Output = TursoOutput;
+
+    fn new() -> Self {
+        Self::default()
+    }
+
+    fn config(&self) -> &Self::Config {
+        self
+    }
+
+    async fn output(
+        self,
+        factory: &mut dyn Factory,
+    ) -> Result<Self::Output, shuttle_service::Error> {
+        match factory.get_environment() {
+            shuttle_service::Environment::Production => {
+                if self.addr.is_empty() {
+                    Err(ShuttleError::Custom(CustomError::msg("missing addr")))
+                } else {
+                    Ok(TursoOutput {
+                        // XXX: should we allow for other kind of connection string ? Not having libsql at
+                        // the start of the url even though the Turso CLI is printing it might be confusing
+                        // instead of just giving the responsability to the user.
```

Comment:
Can we check if the link starts with `libsql://` and if not, try connecting to the url but with the added `libsql://` prefix? In this way, we'll cover both situations I think, unless I haven't understood what you meant.

Reply:
I meant to allow the possibility to connect using an other kind of connection that the `libsql-client` library support, in particular the `reqwest` one, but I don't think there is a reason to support that since Turso's documentation only mention the `libsql` one.